Runtime/Scripts/PeriodicSnapshots.cs

```diff
@@ -86,10 +87,15 @@ private IEnumerator DownloadSnapshots(string timestamp, string path)
 #if UNITY_WEBGL && !UNITY_EDITOR
             var archivePath = FetchArchivePath(timestamp);
             var bytes = File.ReadAllBytes(archivePath);
-            DownloadSnapshot(bytes, bytes.Length, Path.GetFileName(archivePath));
+            DownloadFile(gameObject.name, "OnSnapshotDownloadComplete", Path.GetFileName(archivePath), bytes, bytes.Length);
 #endif
         }
 
+        public void OnSnapshotDownloadComplete(string message)
+        {
+            //Debug.Log("File download complete: " + message);
```

Comment:
Het is dat er toch nog een change nodig is, maar zou je ook in deze class een UnityEvent kunnen toevoegen die hier getriggered wordt? Dat is fijn voor eventuele notificaties (als er geen andere dependency nodig was dan had ik dit niet gevraagd)